samcli/commands/_utils/options.py

```diff
@@ -279,6 +279,20 @@ def no_progressbar_option(f):
     return no_progressbar_click_option()(f)
 
 
+def initial_reporting_sleep_click_option():
+    return click.option(
+        "--initial-reporting-sleep",
```

Comment:
We probably will need a more general option name, I'm also wondering if this could be an env var instead to control behavior in context of where this is being run, for example, you are less likely to run into this (throttling) when running on a personal account, but will be probably run into it more often on accounts which already have high activity.

Reply:
If you want to go that way maybe https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-retries.html#cli-usage-retries-configure can be used as a template?

Reply:
Yeah that would be nice to be consistent with the aws cli. @sriram-mv would that be preferable? Though if you mean a context for enterprise vs personal or isolated might have far great implications on the SAM cli other than just the deployment reporting throttling.
 Otherwise very happy to change the name maybe something like "reporting-delay", "report-wait", "output-waiting", "output-sleep"

Reply:
hey @sriram-mv, can we clarify what is needed to move this PR forward?

Reply:
Hey @sriram-mv, our team stil hits this most days...

Reply:
we're also getting hit with this more and more often I'd love to see this feature ship if even in some beta (changable) state